vulcanai2/plotters/utils.py

```diff
@@ -38,10 +63,10 @@ def __init__(self, network):
     def _hook_top_layers(self):
         def hook_function(module, grad_in, grad_out):
             # TODO: Revisit dim disorder and check isinstance for classes.
-            if module.__class__.__name__ == 'Linear':
+            if isinstance(module, torch.nn.Linear):
                 # grad_in shape is (bias, input, weights)
                 self.gradients = grad_in[1]
-            elif module.__class__.__bases__[0].__name__ == '_ConvNd':
+            elif isinstance(module, torch.nn.modules.conv._ConvNd):
```

Comment:
Just curious, is there any diff wrt to performance by doing this? instead of checking with base Class name?

Reply:
my impression was because it lets you deal with complex class hierarchies

Reply:
I don't get it. What do you mean?

Reply:
👍  thanks